core/src/avm2/globals/number.rs

```diff
@@ -260,6 +260,24 @@ pub fn print_with_radix<'gc>(
         return Value::from(number).coerce_to_string(activation);
     }
 
+    if number.is_nan() {
+        return Ok(AvmString::new_utf8(activation.context.gc_context, "NaN"));
```

Comment:
I may be wrong but doesn't this allocate a new string every time, as opposed to `"NaN".into()` (or `AvmString::from("NaN")` which would not?